src/build/supportedExtensions.js

```diff
@@ -32,42 +30,36 @@ exports.extensions = {
-    { icon: 'composer', extensions: ['composer.lock'], contribType: ctype.filename },
+    { icon: 'composer', extensions: ['composer.json', 'composer.lock'], contribType: ctype.filename },
     { icon: 'cs', extensions: ['cs', 'csx'] },
     { icon: 'cshtml', extensions: ['cshtml'] },
     { icon: 'csproj', extensions: ['csproj'] },
     { icon: 'css', extensions: ['css'] },
-    { icon: 'csslint', extensions: ['csslintrc'] },
+    { icon: 'csslint', extensions: ['.csslintrc'], contribType: ctype.filename },
     { icon: 'cssmap', extensions: ['.css.map'] },
     { icon: 'cucumber', extensions: ['feature'] },
     { icon: 'dartlang', extensions: ['dart'] },
     { icon: 'delphi', extensions: ['pas', 'dfm', 'dpr'] },
     { icon: 'dlang', extensions: ['d'] },
-    { icon: 'docker', extensions: ['dockerfile', 'dockerignore'], contribType: ctype.filename },
-    { icon: 'docker', extensions: ['dockerignore'] },
-    { icon: 'docker', extensions: ['docker-compose.yml'], contribType: ctype.filename },
-    { icon: 'editorconfig', extensions: ['editorconfig'] },
+    { icon: 'docker', extensions: ['dockerfile', '.dockerignore', 'docker-compose.yml'], contribType: ctype.filename },
+    { icon: 'editorconfig', extensions: ['.editorconfig'], contribType: ctype.filename },
     { icon: 'ejs', extensions: ['ejs'] },
     { icon: 'elixir', extensions: ['ex', 'exs', 'eex'] },
     { icon: 'elm', extensions: ['elm'] },
     { icon: 'elm', extensions: ['elm-package.json'], contribType: ctype.filename, contribOrder: 1 },
-    { icon: 'ember', extensions: ['ember-cli'], contribType: ctype.filename },
+    { icon: 'ember', extensions: ['.ember-cli'], contribType: ctype.filename },
     { icon: 'erb', extensions: ['rhtml', 'erb'] },
-    { icon: 'erlang', extensions: ['erl', 'hrl', 'emakefile', 'emakerfile'] },
-    { icon: 'eslint', extensions: ['eslintrc', 'eslintignore'] },
-    { icon: 'eslint', extensions: ['.eslintrc.js'], contribType: ctype.filename },
-    { icon: 'eslint', extensions: ['.eslintrc.json'], contribType: ctype.filename },
-    { icon: 'eslint', extensions: ['.eslintrc.yaml'], contribType: ctype.filename },
-    { icon: 'eslint', extensions: ['.eslintrc.yml'], contribType: ctype.filename },
+    { icon: 'erlang', extensions: ['erl', 'hrl'] },
+    { icon: 'erlang', extensions: ['emakefile', '.emakerfile'], contribType: ctype.filename },
+    { icon: 'eslint', extensions: ['.eslintrc', '.eslintignore', '.eslintrc.js', '.eslintrc.json', '.eslintrc.yaml', '.eslintrc.yml'], contribType: ctype.filename },
     { icon: 'excel', extensions: ['xls', 'xlsx', 'csv', 'ods'] },
     { icon: 'favicon', extensions: ['favicon.ico'], contribType: ctype.filename },
     { icon: 'flash', extensions: ['swf', 'swc', 'sol'] },
-    { icon: 'flow', extensions: ['flowconfig', 'js.flow'] },
+    { icon: 'flow', extensions: ['.flowconfig', 'js.flow'], contribType: ctype.filename },
```

Comment:
`js.flow` it's a complex extension. At least I guess so. I'm not a flow user at all. I think it should be in a separate entry.